code/game/objects/structures/musician.dm

```diff
@@ -27,6 +27,25 @@
 		desc = "This is a space piano, like a regular piano, but always in tune! Even if the musician isn't."
 		icon_state = "piano"
 
+/obj/structure/device/piano/verb/rotate()
+	set name = "Rotate Piano"
+	set category = "Object"
+	set src in oview(1)
+
+	if(config.ghost_interaction)

```

Comment:
This means that if ghost_interaction is on mice, corpses, and restrained people will be able to turn the thing.

My suggestion would be to scrap this if/else, move all of the checks back one indentation, and then add something like 
`
if(istype(usr,/mob/dead/observer) && !config.ghost_interaction))
`
`->return`